### PR TITLE
Make PackageCollectionItem.PackageReferences property value read-only

### PR DESCRIPTION
**Pull request by dtivel**
## Bug

Fixes: https://github.com/NuGet/Home/issues/10338
Regression: No   

## Fix

Details: Making the property value read-only

## Testing/Validation

Tests Added: No  
Reason for not adding tests:  not necessary

CC @sbanni, @rrelyea

### CODE REVIEW

File: src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Common/PackageCollectionItem.cs [hunk 1]
```diff
@@ -19,12 +20,12 @@ public sealed class PackageCollectionItem : PackageIdentity
...
         /// Installed package references.
         /// </summary>
-        public List<IPackageReferenceContextInfo> PackageReferences { get; }
+        public IReadOnlyCollection<IPackageReferenceContextInfo> PackageReferences { get; }
 
         public PackageCollectionItem(string id, NuGetVersion version, IEnumerable<IPackageReferenceContextInfo> installedReferences)
             : base(id, version)
         {
-            PackageReferences = installedReferences?.ToList() ?? new List<IPackageReferenceContextInfo>();
+            PackageReferences = installedReferences?.ToArray() ?? Array.Empty<IPackageReferenceContextInfo>();
```
Comment (sbanni) [inline src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Common/PackageCollectionItem.cs]: Why did you do this change?

Comment (dtivel) [reply src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Common/PackageCollectionItem.cs]: We don't need to allocate a new list if there are no references.

Comment (sbanni) [reply src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Common/PackageCollectionItem.cs]: I meant the ToList -> ToArray change, in general it is better to use ToList over ToArray

Comment (erdembayar) [reply src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Common/PackageCollectionItem.cs]: If I understood correctly Array.Empty doesn't allocate a memory so it doesn't stress GC, is it right?

**Review by erdembayar — APPROVED** [review]

File: src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Common/PackageCollectionItem.cs (hunk 1 above)
Comment (dtivel) [reply src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Common/PackageCollectionItem.cs]: `Array.Empty<T>()` does allocate memory, but it's idempotent, meaning that first call allocates but subsequent calls do not.

**Review by nkolev92 — APPROVED** [review]

**Pull request merged**